### PR TITLE
Fix OGSFX_DEFAULT_SEMANTICS_MAP initialization

### PR DESCRIPTION
**Pull request by fleith**

### CODE REVIEW

**Comment by fleith** [comment]
The problem was related to the `extern` strings like `HW::T_IN_POSITION` that is not initialized before the static map.

File: source/MaterialXGenOgsFx/OgsFxShaderGenerator.cpp [hunk 1]
```diff
@@ -14,49 +14,56 @@ namespace MaterialX
 namespace
 {
     // Semantics used by OgsFx
-    static const StringMap OGSFX_DEFAULT_SEMANTICS_MAP =
+    static StringMap OGSFX_DEFAULT_SEMANTICS_MAP;
```
Comment (bernardkwok) [inline source/MaterialXGenOgsFx/OgsFxShaderGenerator.cpp]: As this isn't really globally required so I would say just move the static map into the method:
OgsFxShaderGenerator::createShader() as that appears to be the only place that it's used.

Comment (fleith) [reply source/MaterialXGenOgsFx/OgsFxShaderGenerator.cpp]: good point, I pushed a new version. Thanks

**Comment by niklasharrysson** [comment]
Thanks Alvaro for fixing my bad code :)

I didn't see your review at first, so I created my own change list for fixing the issue reported by Jonathan. If it's ok with you could we use my fix instead? Partly because it makes the map totally non-static, which could be better for multi-threading support once we get to that, and partly because it simplifies another issues we have in the Maya version of the OgsFx generator where the view direction semantic is overridden to match Maya's semantics.

**Comment by fleith** [comment]
As we already have a fix in the PR https://github.com/autodesk-forks/MaterialX/pull/511 , I'm closing this pull request.

**Pull request closed without merging**